### ISSUE TITLE
Filter Expression can only contain non-primary key attribute

### ISSUE THREAD

**Issue by Sam-Martin** [CloudWanderer-io/CloudWanderer#86]
```
ERROR:cloudwanderer.utils:An error occurred (ValidationException) when calling the Query operation: Filter Expression can only contain non-primary key attributes: Primary key attribute: _resource_type_index
Traceback (most recent call last):
  File "/Users/sammartin/git/CloudWanderer/cloudwanderer/utils.py", line 21, in exception_logging_wrapper
    method(**kwargs)
  File "/Users/sammartin/git/CloudWanderer/cloudwanderer/cloud_wanderer.py", line 106, in write_resources_in_region
    self.write_resources_of_service_in_region(
  File "/Users/sammartin/git/CloudWanderer/cloudwanderer/cloud_wanderer.py", line 141, in write_resources_of_service_in_region
    self.write_resources_of_type_in_region(
  File "/Users/sammartin/git/CloudWanderer/cloudwanderer/cloud_wanderer.py", line 175, in write_resources_of_type_in_region
    self._clean_resources_in_region(service_name, resource_type, region_name, urns)
  File "/Users/sammartin/git/CloudWanderer/cloudwanderer/cloud_wanderer.py", line 201, in _clean_resources_in_region
    storage_connector.delete_resource_of_type_in_account_region(
  File "/Users/sammartin/git/CloudWanderer/cloudwanderer/storage_connectors/dynamodb.py", line 276, in delete_resource_of_type_in_account_region
    for resource in resource_records:
  File "/Users/sammartin/git/CloudWanderer/cloudwanderer/storage_connectors/dynamodb.py", line 239, in read_resources
    result = self.dynamodb_table.query(**query_args)
  File "/Users/sammartin/.pyenv/versions/3.8.6/lib/python3.8/site-packages/boto3/resources/factory.py", line 520, in do_action
    response = action(self, *args, **kwargs)
  File "/Users/sammartin/.pyenv/versions/3.8.6/lib/python3.8/site-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(*args, **params)
  File "/Users/sammartin/.pyenv/versions/3.8.6/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/sammartin/.pyenv/versions/3.8.6/lib/python3.8/site-packages/botocore/client.py", line 676, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the Query operation: Filter Expression can only contain non-primary key attributes: Primary key attribute: _resource_type_index
```

**Comment by Sam-Martin** [comment]
This only occurs when using the experimental concurrency support and is not yet reliably reproducible. 
If anyone has a reproducible scenario, please post it herel

**Comment by Sam-Martin** [comment]
Closing as this seems to have been fixed in release 0.13.0.
